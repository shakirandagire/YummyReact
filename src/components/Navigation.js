import React from 'react';
import { Link } from 'react-router-dom';
import Logout from './logout';


const Navigation = () => (

  <nav className="navbar navbar-expand-lg navbar-light bg-light">
    <a className="navbar-brand"><Link to="/main">Yummy Recipes</Link></a>
    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
      <span className="navbar-toggler-icon" />
    </button>
    <div className="collapse navbar-collapse title" id="navbarNavAltMarkup">
      <div className="navbar-nav">
        <a className="nav-item nav-link active"><Link to="/main">Home </Link><span className="sr-only">(current)</span></a>
        <a className="nav-item nav-link"><Link to="/viewcategories"> Categories </Link></a>
        <a className="nav-item nav-link">Pricing</a>
      </div>
      <Logout />
    </div>
  </nav>

);

export default Navigation;
