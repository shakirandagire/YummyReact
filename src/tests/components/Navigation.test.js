import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Navigation from '../../components/Navigation';

const component = shallow(<Navigation />);
describe('<Navigation/>', () => {
  it('should render properly', () => {
    expect(component.length).toBe(1);
  });

  it('should render navigation', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});

