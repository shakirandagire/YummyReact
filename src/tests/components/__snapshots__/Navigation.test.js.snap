// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> should render navigation 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <a
    className="navbar-brand title"
  >
    <Link
      replace={false}
      to="/"
    >
      Yummy Recipes
    </Link>
  </a>
  <button
    aria-controls="navbarNavAltMarkup"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNavAltMarkup"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse title"
    id="navbarNavAltMarkup"
  >
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link active"
      >
        <Link
          replace={false}
          to="/"
        >
          Home 
        </Link>
        <span
          className="sr-only"
        >
          (current)
        </span>
      </a>
      <a
        className="nav-item nav-link"
      >
        <Link
          replace={false}
          to="/view-categories"
        >
           Categories 
        </Link>
      </a>
    </div>
    <withRouter(Logout) />
  </div>
</nav>
`;
